### ISSUE TITLE
Rule blocked: CORERULES-9554 -Error when using non-string operator values with Match Datasets

### ISSUE THREAD

**Issue by ASL-rmarshall** [cdisc-org/cdisc-rules-engine#738]
If operators that have non-string (e.g., list) values for `value` (such as `is_contained_by`) are used with `Match Datasets`, the following error is reported:

> 'list' object has no attribute 'startswith'

The error occurs because the dataset preprocessor tries to remove domain prefixes from column references in `target` `value`s without accounting for non-string `target` `value`s.

This error can be prevented with a simple change - inserting a type check at [line 79 of dataset_preprocessor.py](https://github.com/cdisc-org/cdisc-rules-engine/blob/main/cdisc_rules_engine/utilities/dataset_preprocessor.py#L79 ):

```python
                    if type(target) == str and target.startswith(f"{domain_name}.")
                       ^^^^^^^^^^^^^^^^^^^^^^^
```

**Comment by ASL-rmarshall** [comment]
I haven't (yet) reported this as blocking a rule because I have used a workaround for using `is_contained_by` with `Match Datasets` - in DDF00037, I have used `any` with `equal_to` instead.

**Comment by ASL-rmarshall** [comment]
I've now updated this issue to indicate that it's blocking a rule. In fact, it is blocking multiple rules:

- DDF00052: https://jira.cdisc.org/browse/CORERULES-9554
- DDF00053: https://jira.cdisc.org/browse/CORERULES-9555
- DDF00054: https://jira.cdisc.org/browse/CORERULES-9556
- DDF00055: https://jira.cdisc.org/browse/CORERULES-9557
- DDF00056: https://jira.cdisc.org/browse/CORERULES-9558
- DDF00057: https://jira.cdisc.org/browse/CORERULES-9559
- DDF00058: https://jira.cdisc.org/browse/CORERULES-9560

I have tried a workaround of using the `record_count` operation instead of `is_not_unique_set`, but the problem still occurs when try to compare the result of `record_count` with an integer.

**Comment by ASL-rmarshall** [comment]
Also blocking DDF00069: https://jira.cdisc.org/browse/CORERULES-9604
